src/lib.rs

```diff
@@ -766,28 +763,18 @@ impl SystemHandle {
         &self,
         recipient: Recipient<M>,
     ) -> Result<(), SendError> {
-        let mut event_subscribers = self.event_subscribers.write();
+        // Subscribe first. We assume it is better to receive an event twice rather than not at all.
```

Comment:
This is weird assumption for me. With the previous logic, we can only miss an event if it is published right in between us checking the cache and adding the callback to event_subscribers. That is a very short window and at that point the notion of "last published event" is unclear: from POV of the user, that missed event was published *after* they called `subscribe_and_receive_latest()` so it's expected the subscriber won't receive it.

We can't assume it's ok to deliver events twice - that depends on the semantics of the subscriber, not the event itself.

I think we should send cached first and subscribe later.